src/mbgl/tile/geometry_tile_data.cpp

```diff
@@ -17,71 +17,93 @@ static double signedArea(const GeometryCoordinates& ring) {
     return sum;
 }
 
-static ClipperLib::Path toClipperPath(const GeometryCoordinates& ring) {
-    ClipperLib::Path result;
+static LinearRing<std::int32_t> toWagyuPath(const GeometryCoordinates& ring) {
+    LinearRing<std::int32_t> result;
     result.reserve(ring.size());
     for (const auto& p : ring) {
         result.emplace_back(p.x, p.y);
     }
     return result;
 }
 
-static GeometryCoordinates fromClipperPath(const ClipperLib::Path& path) {
-    GeometryCoordinates result;
-    result.reserve(path.size() + 1);
-
-    for (const auto& p : path) {
-        using Coordinate = GeometryCoordinates::coordinate_type;
-        assert(p.x >= std::numeric_limits<Coordinate>::min());
-        assert(p.x <= std::numeric_limits<Coordinate>::max());
-        assert(p.y >= std::numeric_limits<Coordinate>::min());
-        assert(p.y <= std::numeric_limits<Coordinate>::max());
-        result.emplace_back(Coordinate(p.x), Coordinate(p.y));
-    }
-
-    // Clipper does not repeat initial point, but our geometry model requires it.
-    if (!result.empty()) {
-        result.push_back(result.front());
-    }
-
-    return result;
+static void pushWagyuRing(GeometryCollection & solution,
+                          mapbox::geometry::wagyu::ring_ptr<std::int32_t>& r) {
+    GeometryCoordinates lr;
+    lr.reserve(r->size() + 1);
+    auto firstPt = r->points;
+    auto ptIt = r->points;
+    do {
+        lr.emplace_back(ptIt->x, ptIt->y);
+        ptIt = ptIt->prev;
+    } while (ptIt != firstPt);
```

Comment:
@flippmoke  is there a speed advantage to `do/while` loop instead of a c++11 for loop over the `ring_ptr` (or would that be possible?)